tests/test_interactions.py

```diff
@@ -340,3 +341,120 @@ def test_cyin007_click_twice(dash_duo):
 
     assert clicked_label_1 == clicked_label_2
     assert clicked_timestamp_1 != clicked_timestamp_2
+
+
+def test_cyin008_ctx_menu_remove_node(dash_duo):
+    init_pos, actions, _ = create_app(dash_duo)
+    init_x, init_y = init_pos["Node 1"]
+
+    # Open the Drag data JSON tab
+    actions.move_to_element(dash_duo.find_element("#tabs > div:nth-child(5)"))
+    actions.click().perform()
+    time.sleep(1)
+
+    # Select the JSON output element before removal
+    elements = dash_duo.find_element("pre#elements-data-json-output")
+    nb_elements_before = len(json.loads(elements.text))
+
+    # move mouse to first node and right click and click on remove
+    actions.move_to_location(init_x, init_y)
+    actions.context_click()
+    actions.move_to_element(dash_duo.find_element("button#remove"))
+    actions.click()
+    actions.perform()
+    time.sleep(1)
+
+    # Select the JSON output element after removal
+    elements = dash_duo.find_element("pre#elements-data-json-output")
+    nb_elements_after = len(json.loads(elements.text))
+
+    assert nb_elements_before > nb_elements_after
+
+
+def test_cyin009_ctx_menu_remove_edge(dash_duo):
+    _, actions, edge_positions = create_app(dash_duo)
+    print(edge_positions)
+    init_x, init_y = edge_positions["Edge from Node 1 to Node 2"]
+
+    # Open the Drag data JSON tab
+    actions.move_to_element(dash_duo.find_element("#tabs > div:nth-child(5)"))
+    actions.click().perform()
+    time.sleep(1)
+
+    # Select the JSON output element before removal
+    elements = dash_duo.find_element("pre#elements-data-json-output")
+    nb_elements_before = len(json.loads(elements.text))
+
+    # move mouse to an edge and right click and click on remove
+    actions.move_to_location(init_x, init_y)
+    actions.context_click()
+    actions.move_to_element(dash_duo.find_element("button#remove"))
+    actions.click()
+    actions.perform()
+    time.sleep(1)
+
+    # Select the JSON output element after removal
+    elements = dash_duo.find_element("pre#elements-data-json-output")
+    nb_elements_after = len(json.loads(elements.text))
+
+    assert nb_elements_before - 1 == nb_elements_after
```

Comment:
To be super nitpicky, it would probably be *best* to actually count the number of edges rather than just elements. So that you could distinguish between an edge being removed vs. a node being removed, say. I don't necessarily object to leaving it like this though as it's a minor difference. (same goes for the other tests)